pkg/controller/noderesource/node_resource_controller.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesource
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	predictionapi "github.com/gocrane/api/prediction/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/client-go/scale"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+)
+
+// NodeResourceReconciler reconciles a NodeResource object
+type NodeResourceReconciler struct {
+	client.Client
+	Scheme 		*runtime.Scheme
+	Log         logr.Logger
+	RestMapper  meta.RESTMapper
+	Recorder    record.EventRecorder
+	scaleClient scale.ScalesGetter
```

Comment:
scaleClient is not used in your controller

Reply:
no use i will remove it